tests/network/test_vif_allowed_ip.py

```diff
@@ -0,0 +1,87 @@
+import ipaddress
+import os
+import pytest
+
+# Requirements:
+# - one XCP-ng host (--host) >= 8.2 (>= 8.3 for the CIDR tests)
+# - a VM (--vm)
+
+def ip_responsive(ip):
+    return not os.system(f"ping -c 3 -W 10 {ip} > /dev/null 2>&1")
+
+@pytest.mark.small_vm
+class Test:
+    def test_unallowed_ip(self, running_vm):
+        vm = running_vm
+        # FIXME: make sure IP and VIF are related.
+        vif = vm.vifs()[0]
+        ip = vm.ip
+        ip_address = ipaddress.ip_address(ip)
+        ip_family = ip_address.version
+        dummy_ip = str(ip_address + 1)
+        assert ip_responsive(ip)
+
+        vif.param_add(f"ipv{ip_family}-allowed", dummy_ip)
+        vif.param_set("locking-mode", "locked")
+        assert not ip_responsive(ip)
+
+        vif.param_clear(f"ipv{ip_family}-allowed")
+        vif.param_set("locking-mode", "unlocked")
+
+    def test_allowed_ip(self, running_vm):
+        vm = running_vm
+        # FIXME: make sure IP and VIF are related.
+        vif = vm.vifs()[0]
+        ip = vm.ip
+        ip_address = ipaddress.ip_address(ip)
+        ip_family = ip_address.version
+
+        vif.param_clear(f"ipv{ip_family}-allowed")
+        vif.param_set("locking-mode", "locked")
+        assert not ip_responsive(ip)
+
+        vif.param_add(f"ipv{ip_family}-allowed", ip)
+        assert ip_responsive(ip)
+
+        vif.param_remove(f"ipv{ip_family}-allowed", ip)
+        assert not ip_responsive(ip)
+
+        vif.param_set("locking-mode", "unlocked")
+
+    def test_unallowed_cidr(self, host_at_least_8_3, running_vm):
```

Comment:
The host_at_least_8_3 fixture is not used in the function, and marking it explicitly with pytest makes it easier to spot. This would be consistent with other tests.

Alternatively, we could move these two tests to a separate class and mark the whole class with this fixture, rather than the test functions individually.

```suggestion
    @pytest.mark.usefixtures("host_at_least_8_3")
    def test_unallowed_cidr(self, running_vm):
```

Reply:
i'll split in 2 classes :)